File: Neon Number/Neon.cpp
#include <iostream>
using namespace std;
 
int main() {
   int n,sq,sum = 0;
   cout<<"Enter the number: ";
   cin>>n;
   sq = n * n;//find square of number
   /*calculating the sum of square*/
   while(sq > 0)
   {
       sum = sum + sq % 10;
       sq = sq / 10;
   }
   if(sum == n)
   {
       cout<<n<<" is a neon number.";
   }
   else
   {
       cout<<n<<" is not a neon number.";
   }
   return 0;
}

File: FunnyAlgorithms-master/Neon Number/neon.cs
using System;

namespace ConsoleApp
{
    public class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("Enter the number: ");
            var inputNumber = Console.ReadLine();
            if (int.TryParse(inputNumber, out int number))
            {
                Console.WriteLine(IsNeonNumber(Convert.ToInt32(number)));
            }
            else
            {
                Console.WriteLine("Please enter numeric value.");
            }
            Console.ReadLine();
        }

        static string IsNeonNumber(int x)
        {
            // storing the square of x 
            int square = x * x;

            // calculating the sum of digits of square
            int sumOfDigits = 0;
            while (square != 0)
            {
                sumOfDigits += square % 10;
                square /= 10;
            }

            return $"{x} is " + ((sumOfDigits == x) ? string.Empty : "not ") + "neon number";
        }
    }
}
